File: c++/Inheritance/InheritData.cpp
//Inheritance of Data
//refer : Inheitance.txt
//see : InheritData.png

#include<iostream>

using namespace std;

class A
{
private :
  int x;
protected:
  int y;
public:
  int z;

  A();
  A(int,int,int);

  void display();
  int getX(); //accessor (getter)
};

A :: A()
{
  cout<<"\n A A()";
  x = 10;
  y = 20;
  z = 30;
}

A :: A(int q, int w , int e)
{
  cout<<"\n A A(int,int,int)";
  x = q;
  y = w;
  z = e;
}

void A :: display()
{
  cout<<"\n "<<x<<" "<<y<<" "<<z;
}

int A :: getX()
{
  return x;
}

class B : public A
{
protected:
  int sum;
public:
  B();
  B(int,int,int);

  void display();//redefine the inherited method (override)
};

B::B() // : A()
{
  cout<<"\n B B()";
  sum = getX()+y+z;
}

B::B(int p, int q, int r):A(p,q,r)
{
  cout<<"\n B B(int,int,int)";
  sum = getX()+y+z;
}


void B :: display()
{
  cout<<"\n"<<getX() <<" + "<< y <<" + "<<z <<" = "<<sum;
}

int main()
{
  B objB(1,2,3);
  objB.display();
  return 0;
}

File: Java/Inheritance/InheritData.java
/*
Inheritance of data

Refer: Inheritance.txt
See: InheritData.png
*/

class A
{
 //non static data members
 int x, y;

 //non static initializer block
 {
   System.out.println("NSIB of A"); 
 } 


 A()
 {
  System.out.println("A()");
  x = 10;//preset data
  y = 20;//preset data
 }

 A(int q, int w)
 {
  System.out.println("A(int,int)");
  x = q;//parameterized data
  y = w;//parameterized data
 }

 void display()
 {
  System.out.println(x + " " + y);
 }
 
}//A

class B extends A
{
 //non static data member
 int z;

 //non static initializer block
 {
   System.out.println("NSIB of B"); 
 } 


 B()//:A()
 {
  //super();
  System.out.println("B()");
  z = x+y;  
 }

 B(int q, int w)//:A(q,w)
 {
  //must be the first statement in the constructor of sub class
  super(q, w); //passing parameters to A(int,int)
  System.out.println("B(int,int)");
  z = x+y;  
 }

 //override
 void display()
 {
  System.out.println(x + " + " + y + " = " + z);
 }
}//B

class InheritData
{
 public static void main(String args[])
 {

  //Create an object of the sub class
  //(application of inheritance of data)
  
  //B objB = new B();
  B objB = new B(1,2);
  objB.display();
 }
}